chemprop/v2/cli/train.py

```diff
@@ -32,41 +32,308 @@ class TrainSubcommand(Subcommand):
+        "--empty_cache",
+        action="store_true",
+        help="Whether to empty all caches before training or predicting. This is necessary if multiple jobs are run within a single script and the atom or bond features change.",
+    )
+    parser.add_argument(
+        "--cache_cutoff",
+        type=float,
+        default=10000,
+        help="Maximum number of molecules in dataset to allow caching. Below this number, caching is used and data loading is sequential. Above this number, caching is not used and data loading is parallel. Use 'inf' to always cache.",
+    )
+    parser.add_argument(
+        "--constraints_path",
+        type=str,
+        help="Path to constraints applied to atomic/bond properties prediction.",
+    )
+
+    # to do: see if we need to add functions from CommonArgs 
+
+def add_train_args(parser: ArgumentParser) -> ArgumentParser:
     parser.add_argument(
         "-i",
         "--input",
-        "--data-path",
-        help="path to an input CSV containing SMILES and associated target values",
+        "--data_path",
+        type=str,
+        help="Path to an input CSV containing SMILES and associated target values.",
     )
-    parser.add_argument("-o", "--output-dir")
     parser.add_argument(
-        "--logdir",
-        nargs="?",
-        const="chemprop_logs",
-        help="runs will be logged to {logdir}/chemprop_{time}.log. If unspecified, will use 'output_dir'. If only the flag is given (i.e., '--logdir'), then will write to 'chemprop_logs'",
+        "-o",
+        "--output-dir",
+        "--save_dir",
+        type=str,
+        help="Directory where model checkpoints will be saved.",
+    )
+    parser.add_argument(
+        "--log_frequency",
+        type=int,
+        default=10,
+        help="The number of batches between each logging of the training loss.",
+    )
+    parser.add_argument(
+        "--checkpoint_frzn",
+        type=str,
+        help="Path to model checkpoint file to be loaded for overwriting and freezing weights."
+    )
+    parser.add_argument(
+        "--frzn_ffn_layers",
+        type=int,
+        default=0,
+        help="Overwrites weights for the first n layers of the ffn from checkpoint model (specified checkpoint_frzn), where n is specified in the input. Automatically also freezes mpnn weights.",
+    )
+    parser.add_argument(
+        "--freeze_first_only",
+        action="store_true",
+        help="Determines whether or not to use checkpoint_frzn for just the first encoder. Default (False) is to use the checkpoint to freeze all encoders. (only relevant for number_of_molecules > 1, where checkpoint model has number_of_molecules = 1)",
+    )
+    parser.add_argument(
+        "--test",
+        action="store_true",
+        help="Whether to skip training and only test the model.",
+    )
+    parser.add_argument(
+        "-q",
+        "--quiet",
+        action="store_true",
+        help="Skip non-essential print statements.",
+    )
```

Comment:
delete

Reply:
We should delete this because we don't plan to implement a quiet option? I feel a quiet option is a large developer burden and agree to delete this option.

Reply:
this should be controlled by a `verbosity` argument which globally sets the logging level and then each individual module should log accordingly.